actors/builtin/miner/miner_actor.go

```diff
@@ -2021,12 +2054,27 @@ func requestCurrentTotalPower(rt Runtime) *power.CurrentTotalPowerReturn {
 // may be slightly lower than the true amount.
 // Computing vesting here would be almost always redundant since vesting is quantized to ~daily units.
 // Vesting will be at most one proving period old if computed in the cron callback.
-func verifyPledgeMeetsInitialRequirements(rt Runtime, st *State) {
+func verifyPledgeRequirementsAndRepayDebts(rt Runtime, st *State) abi.TokenAmount {
+	currBalance := rt.CurrentBalance()
+	unlockedBalance := st.GetUnlockedBalance(currBalance)
+	if st.FeeDebt.GreaterThan(unlockedBalance) {
+		rt.Abortf(exitcode.ErrInsufficientFunds,
+			"unlocked balance can not repay fee debt (%v < %v)",
+			unlockedBalance, st.FeeDebt,
+		)
+	}
+	debtToRepay := st.FeeDebt
+	st.FeeDebt = big.Zero()
+	currBalance = big.Sub(currBalance, debtToRepay)
+	unlockedBalance = big.Sub(unlockedBalance, debtToRepay)
+	AssertMsg(currBalance.GreaterThanEqual(big.Zero()), "debt repayment can not take balance below zero")
+
 	if !st.MeetsInitialPledgeCondition(rt.CurrentBalance()) {
```

Comment:
Blocking: Using `rt.CurrentBalance()` after the calculations above looks like a bug.

This is what I'm getting at with the requests for tests. We need to test _this_ code directly, not as part of complicated whole actor flow tests.

This `verifyPledgeRequirementsAndRepayDebts` probably deserves to be elevated to something we can test directly, maybe broken out to a `fees.go` file (or into `monies.go`) along with the things it calls.

As a structural note, we're trying as hard as we can not to manipulate the state objects from actor code, except through methods, so they can enforce their own invariants. The setting of `st.FeeDebt` here goes against that a little. I would like to see structure that would let us replace this method with something like:
```
toBurn, err := st.RepayDebt(rt.CurrentBalance())
// if err ...
Assert(st.FeeDebt == big.Zero)
if (rt.CurrentBalance() - toBurn).LessThan(st.InitialPledgeRequirement) {
...
}

```